vdirsyncer/http.py

```diff
@@ -37,10 +37,10 @@ def prepare_auth(auth, username, password):
     if username and password:
         if auth == 'basic' or auth is None:
             return (username, password)
-        elif auth == 'digest':
+        if auth == 'digest':
             from requests.auth import HTTPDigestAuth
             return HTTPDigestAuth(username, password)
-        elif auth == 'guess':
+        if auth == 'guess':
```

Comment:
I don't really like this change.

It makes it far harder to figure out the flow. At a glance, I'd get the impression that this code run `if auth == 'guess'`, but there's actually a return nested further above.

The `elif` makes it a bit more obvious that these are mutually exclusive code-paths.

Reply:
In all programming languages, writing `else` after `return` has no added value.

Reply:
It helps readability, IMO.